scripts/migrate_1822_pnw_merger.rb

```diff
@@ -0,0 +1,389 @@
+
+  middle_block = (first_block.last + 1)..(second_block.first - 1)
+  start_idx = actions[first_block.first]['id']
+  reordered = actions[second_block] + actions[middle_block] + actions[first_block]
+  reordered.each_with_index { |action, i | action['id'] = start_idx + i }
+  actions[first_block.first..second_block.last] = reordered
+end
+
+# Moves an auto_actions array from one action to another. If both actions have
+# auto_actions arrays then they will be switched. If neither have auto_actions
+# then nothing is changed.
+# @param actions [Array] Array of actions.
+# @param from [Integer] Index of the action with the auto_actions.
+# @param to [Integer] Index of the action to move the auto_actions to.
+def move_auto_actions(actions, from, to)
+  from, to = [from, to].minmax # In case from > to
+  first = actions[from]
+  second = actions[to]
+  auto_actions1 = first['auto_actions']
+  auto_actions2 = second['auto_actions']
+  return if auto_actions1.nil? && auto_actions2.nil?
+
+  if auto_actions1.nil?
+    second.delete('auto_actions')
+  else
+    second['auto_actions'] = auto_actions1
+  end
+  if auto_actions2.nil?
+    first.delete('auto_actions')
+  else
+    first['auto_actions'] = auto_actions2
+  end
+end
+
+# If inserting/deleting actions, modify the given `actions` and return `nil`
+#
+# If editing existing actions, modify them on `actions` in place, and return an
+# array containing the just the modified actions (so the modified actions will
+# be in both the originally given `actions` and in the returned array)
+def repair(game, original_actions, actions, broken_action, data, pry_db: false)
+  optionalish_actions = %w[message buy_company]
+  broken_action_idx = actions.index(broken_action)
+  action = broken_action['original_id'] || broken_action['id']
+  step = game.active_step
+  prev_actions = actions[0..broken_action_idx - 1]
+  prev_action = prev_actions[prev_actions.rindex { |a| !optionalish_actions.include?(a['type']) }]
+  next_actions = actions[broken_action_idx + 1..]
+  next_action = next_actions.find { |a| !optionalish_actions.include?(a['type']) }
+  current_entity = step.current_entity
+
+  entity_id = broken_action['entity']
+  entity = game.corporation_by_id(entity_id) ||
+           game.company_by_id(entity_id) ||
+           game.player_by_id(entity_id)
+
+  step_actions = step.actions(current_entity)
+
+  ################
+  # BEGIN REPAIR #
+  ################
+  # When a new migration is needed for something more than adding/removing pass
+  # actions, delete blocks here for completed migrations and add a new commented
+  # block; see the history of this file for examples of previous migrations.
+
+  if pry_db
+    require 'pry-byebug'
+    binding.pry
+  end
+
+  # Generic handling for when a change just needs pass actions to be
+  # inserted/deleted
+
+  # try deleting auto_action pass
+  if game.round.actions_for(game.current_entity).include?(broken_action['type']) &&
```

Comment:
This is the only change in this file from the normal `migrate_game.rb`

The check for the current action entities matching is removed, maybe it would be fine to remove in the base on as well.